react-app/src/components/table/DataTable.tsx

```diff
@@ -182,12 +190,99 @@ type FilterSearchDataGridProps = {
   const [selectValue, setSelectValue] = useState<string>(props.defaultFilter);
   const [isExporting, setIsExporting] = useState<boolean>(false);
+  const [dataSourceLoading, setDataSourceLoading] = useState<boolean>(false);
   const tableApiRef = useGridApiRef(); // Ref to MUI DataGrid
+  const previousController = useRef<AbortController>();
+
+  interface ITableModelCollection {
+    pagination?: GridPaginationModel;
+    sort?: GridSortModel;
+    filter?: GridFilterModel;
+    quickFilter?: string[];
+  }
+
+  const dataSourceUpdate = (models: ITableModelCollection) => {
+    const { pagination, sort, filter, quickFilter } = models;
+    if (previousController.current) {
+      previousController.current.abort();
+    }
+    //We use this AbortController to cancel requests that haven't finished yet everytime we start a new one.
+    const controller = new AbortController();
+    const signal = controller.signal;
+    previousController.current = controller;
+    let sortObj: { sortKey?: string; sortOrder?: string; sortRelation?: string } = {};
+    if (sort?.length) {
+      sortObj = { sortKey: sort[0].field, sortOrder: sort[0].sort };
+    }
+    const filterObj = {};
+    if (filter?.items) {
+      for (const f of filter.items) {
+        const asCamelCase = f.field.charAt(0).toLowerCase() + f.field.slice(1);
+        if (f.value != undefined) {
+          filterObj[asCamelCase] = `${f.operator},${f.value}`;
+        } else if (f.operator === 'isNotEmpty' || f.operator === 'isEmpty') {
+          filterObj[asCamelCase] = f.operator;
+        }
+      }
+    } else if (quickFilter) {
+      const keyword = quickFilter[0];
+      for (const fieldName of tableApiRef.current.getAllColumns().map((col) => col.field)) {
+        if (keyword != undefined) {
+          const asCamelCase = fieldName.charAt(0).toLowerCase() + fieldName.slice(1);
+          filterObj[asCamelCase] = `contains,${keyword}`;
+        }
+      }
+    }
+    setDataSourceLoading(true);
+    props
+      .dataSource(
+        {
+          quantity: pagination.pageSize,
+          page: pagination.page,
+          ...sortObj,
+          ...filterObj,
+        },
+        signal,
+      )
+      .then((resolved) => {
+        setDataSourceRows(resolved);
+      })
+      .catch((e) => {
+        if (!(e instanceof DOMException)) {
+          //Represses DOMException which is the expected result of aborting the connection.
+          //If something else happens though, we may want to rethrow that.
+          throw e;
+        }
+      })
+      .finally(() => {
+        setDataSourceLoading(false);
+      });
+  };
+
+  useEffect(() => {
+    if (props.dataSource) {
+      dataSourceUpdate(tableModel);
+    }
+  }, [tableModel]);
+
+  useEffect(() => {
+    if (props.initialState?.sorting?.sortModel) {
+      tableApiRef.current.setSortModel(props.initialState.sorting.sortModel);
+    }
+  }, []); //Empty array makes this fire only on first render.
```

Comment:
Is this not handled by the initialState prop on the DataGrid?

Reply:
Was running into issues with the sort model changing unexpectedly before but I must have corrected this somewhere since it seems like this isn't required anymore